bin/sdrfToNfConf.R

```diff
@@ -1077,14 +1077,14 @@ configs <- lapply(species_list, function(species){
         }else{
 
           # For each library we check if there is a fastq URI that can supply the file
-          uri_select <- apply(species.protocol.sdrf[,uri_cols], 2, function(x) basename(x) == files)         
+          uri_select <- apply(species.protocol.sdrf[,uri_cols], 2, function(x) basename(x) == files)
           
           if (is.data.frame(uri_select) || is.matrix(uri_select)) {
             # the following line gives an error: dim(X) must have a positive length
             # if not a matrix or a data.frame
             missing_uri_files <- files[which(! apply(apply(species.protocol.sdrf[,uri_cols], 2, function(x) basename(x) == files), 1, any))]
           } else {
-            missing_uri_files <- files[which(! apply(as.data.frame( apply(species.protocol.sdrf[,uri_cols], 2, function(x) basename(x) == files) ), 1, any))]
+            missing_uri_files <- files[which(! any(apply(species.protocol.sdrf[, uri_cols], 2, function(x) basename(x) == files))  )]
```

Comment:
Can you please explain briefly what are you trying to do here.. So I know what I understood from code is right..

Reply:
Hi @anilthanki  Did you have a look to the description of the PR?

Basically,  in this chunk for each library it is checked if there is a fastq URI that can supply the file. The error was in the fix for the case when `uri_select` is not a matrix or a data.frame